src/Model/ROI.py

```diff
@@ -856,3 +860,110 @@ def renumber_roi_number(sequence):
             item.add_new(Tag("ReferencedROINumber"), "IS", str(roi_number))
         roi_number += 1
     return sequence
+
+
+def roi_to_geometry(dict_rois_contours):
+    """
+    Convert ROI contour data in each image slice to a geometry object
+    :param dict_rois_contours: A dictionary with key-value pair {slice-id: contour sequence}
+    :return: A dictionary with key-value pair {slice-id: Polygon/Multipolygon Object}
+    """
+    roi_contour_sequence_geometry = {}
+
+    for slice_id, contour_sequence in dict_rois_contours.items():
+        multi = bool(len(contour_sequence) - 1)
+
+        # Generate geometry. Handling exceptions for invalid contour data, multiple polygons
+        if not multi:
+            contour_data = contour_sequence[0]
+            try:
+                polygon = Polygon(contour_data)
+                roi_contour_sequence_geometry[slice_id] = polygon
+            except (TopologicalError, ValueError):
+                polygon = Polygon()
+                roi_contour_sequence_geometry[slice_id] = polygon
+        else:
+            polygon_list = []
+            for contour_data in contour_sequence:
+                try:
+                    sub_polygon = Polygon(contour_data)
+                    polygon_list.append(sub_polygon)
+                except (TopologicalError, ValueError):
+                    sub_polygon = Polygon()
+                    polygon_list.append(sub_polygon)
+            polygon = MultiPolygon(polygon_list)
+            roi_contour_sequence_geometry[slice_id] = polygon
+    return roi_contour_sequence_geometry
+
+
+def manipulate_rois(first_sequence_geometry, second_sequence_geometry, image_uids, operation):
+    """
+    Compute the intersection of two ROIs
+    :param first_sequence_geometry: The geometry dictionary of the first ROI
+    :param second_sequence_geometry: The geometry dictionary of the second ROI
+    :param image_uids: The list of all image uids
+    :param operation: A string of either "INTERSECTION", "UNION", or "DIFFERENCE"
+    :return: A dictionary with key-value pair {slice-id: Polygon/Multipolygon Object}
+    """
+    if operation not in ["INTERSECTION", "UNION", "DIFFERENCE"]:
```

Comment:
please create a set allowed_polygon, or if you use a dictionary of operation function for each operation like the comment below, then it should be a dictionary

Reply:
Done